python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -285,6 +285,67 @@ def test_parallel_writes_and_appends_index_validation(arctic_library, finalize_m
         expected = pd.concat([df_0, df_1, df_2]) if finalize_method == StagedDataFinalizeMethod.APPEND else pd.concat([df_1, df_2])
         assert_frame_equal(received, expected)
 
+@pytest.mark.parametrize("finalize_method", (StagedDataFinalizeMethod.APPEND, StagedDataFinalizeMethod.WRITE))
+class TestFinalizeWithEmptySegments:
+    def test_staged_segment_is_only_empty_dfs(self, lmdb_library, finalize_method):
+        lib = lmdb_library
+        lib.write("sym", pd.DataFrame([]), staged=True)
+        lib.write("sym", pd.DataFrame([]), staged=True)
+        lib.finalize_staged_data("sym", mode=finalize_method)
+        assert_frame_equal(lib.read("sym").data, pd.DataFrame([], index=pd.DatetimeIndex([])))
+
+    def test_staged_segment_has_empty_df(self, lmdb_library, finalize_method):
+        lib = lmdb_library
+        index = pd.DatetimeIndex([pd.Timestamp(2024, 1, 1), pd.Timestamp(2024, 1, 3), pd.Timestamp(2024, 1, 4)])
+        df1 = pd.DataFrame({"col": [1, 2, 3]}, index=index)
+        df2 = pd.DataFrame({})
+        df3 = pd.DataFrame({"col": [4]}, index=pd.DatetimeIndex([pd.Timestamp(2024, 1, 5)]))
+        lib.write("sym", df1, staged=True)
+        lib.write("sym", df2, staged=True)
+        lib.write("sym", df3, staged=True)
+        lib.finalize_staged_data("sym", mode=finalize_method)
+        assert_frame_equal(lib.read("sym").data, pd.concat([df1, df2, df3]).sort_index())
```

Comment:
Why does this need `sort_index`?

Reply:
It's a leftover I had a case with overlapping segments. It's not needed anymore.